rdr_service/storage.py

```diff
@@ -68,7 +74,12 @@ def _get_blob_name_from_local_path(self, local_path):
         return blob_name
 
     def open(self, path, mode):
-        return open(self._get_local_path(path), mode)
+        local_path = self._get_local_path(path)
+        directory = os.path.dirname(local_path)
+        if directory:
+            pathlib.Path(directory).mkdir(parents=True, exist_ok=True)
+
```

Comment:
Why do we create the directory before we open a file in it?

Reply:
to ensure parents exist or are created. Some tests pass in a hierarchy and this goes to temp.

Reply:
@rennat  I don't think we need create the directory when we open a file. If the file is not there, even you create the directory, the file still not there. I didn't get the intention of this part. What do you think?

Reply:
You are right that we do not need to when opening for reading. However, we do need to create the directory when opening for writing.
We use the filesystem storage provider only for local development so this shortcut seemed reasonable. To get a `complete` solution for this we would need a file-wrapper that only creates the directories when `write` is called. This seemed like a lot of extra work only to fix a minor inconvenience for local development.

summary: yes it is a problem but it has minimal impact and it is not worth the time to fix it properly right now

Reply:
OK, I got you. When I do my test cases, I manually create the directories before I open them, so that's why I didn't need this part.